root.go

```diff
@@ -85,19 +85,23 @@ func IsFile(fsn FSNode) bool {
 
 // Root represents the root of a filesystem tree.
 type Root struct {
-
 	// Root directory of the MFS layout.
 	dir *Directory
 
 	repub *Republisher
+
+	ctx    context.Context
+	cancel context.CancelFunc
```

Comment:
I think this is a misuse of a context.
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

-- [golang docs](https://golang.org/pkg/context/)

unless the struct is used as a parameter, according to Brad Fitzpatrick

It also makes it confusing about what is the correct way to shutdown msf.  Call `Close`, cancel the context, both?  Better to have one or the other.  How about using a context _only_ to cancel a function call that may take too long to return, and using `Close` to shutdown mfs root.